### PR TITLE
Fixes #26106 - Refactor Autocomplete Redux

### PR DESCRIPTION
**Pull request by Ron-Lavi**
Part of the code in [filter's forms Autocomplete](https://github.com/theforeman/foreman/pull/6142)
Enable to consume the AutoComplete component as an independent component without the SearchBar.

@theforeman/ui-ux could you have a look?
@amirfefer needs this also for his PR in report template feature

### CODE REVIEW

**Comment by Ron-Lavi** [comment]
What have been done in this PR:
- Rename `Autocomplete/index.js` into `Autocomplete/Autocomplete.js`.
- Add a new `Autocomplete/index.js` which is the connect wrapper for Redux.
- Delete unnecessary keys from SearchBar's `mapStateToProps`
- Delete unnecessary props from `SearchBar/SearchBar.js`.
- Update integration tests on both components.
- Update all imported paths.

**Comment by Laviro** [comment]
Errors seem unrelated

File: webpack/assets/javascripts/react_app/components/AutoComplete/AutoComplete.js [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+      </div>
+    );
+  }
+}
+
+AutoComplete.propTypes = {
+  results: PropTypes.array,
+  searchQuery: PropTypes.string,
+  initialQuery: PropTypes.string,
+  inputProps: PropTypes.object,
```
Comment (amirfefer) [inline webpack/assets/javascripts/react_app/components/AutoComplete/AutoComplete.js:208]: can you please use 'shape' ?

File: webpack/assets/javascripts/react_app/components/SearchBar/SearchBar.js [hunk 2]
```diff
@@ -4,71 +4,40 @@ import PropTypes from 'prop-types';
 import URI from 'urijs';
 import AutoComplete from '../AutoComplete';
 import Bookmarks from '../bookmarks';
-import { noop } from '../../common/helpers';
 
-const handleSearch = (searchQuery, onSearch) => {
-  onSearch(searchQuery);
```
Comment (amirfefer) [inline webpack/assets/javascripts/react_app/components/SearchBar/SearchBar.js:10]: why did you remove `onSearch` option?

File: webpack/assets/javascripts/react_app/components/SearchBar/__tests__/__snapshots__/integration.test.js.snap [hunk 3]
```diff
@@ -49,7 +9,7 @@ Object {
           "controller": "models",
           "searchQuery": "result",
           "status": "PENDING",
-          "trigger": "ITEM_SELECT",
+          "trigger": "INPUT_CHANGE",
```
Comment (amirfefer) [inline webpack/assets/javascripts/react_app/components/SearchBar/__tests__/__snapshots__/integration.test.js.snap]: why the trigger has been changed to `INPUT_CHANGE`?

**Review by amirfefer — CHANGES_REQUESTED** [review]
Thanks @Laviro ! 
Tested and it works well

just few minor issues

File: webpack/assets/javascripts/react_app/components/AutoComplete/AutoComplete.js (hunk 1 above)
Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/components/AutoComplete/AutoComplete.js:208]: These props are the input element attributes, I can't really mention all.. tried to set the prop to an html node but it throws an error.

File: webpack/assets/javascripts/react_app/components/SearchBar/SearchBar.js (hunk 2 above)
Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/components/SearchBar/SearchBar.js:10]: I think there isn't really a use of it in the SearchBar

File: webpack/assets/javascripts/react_app/components/SearchBar/__tests__/__snapshots__/integration.test.js.snap (hunk 3 above)
Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/components/SearchBar/__tests__/__snapshots__/integration.test.js.snap]: Thanks, actually there was no need for this snapshot even before,
the submit button, triggers the `turbolinks.visit()` which is being tested in this file too,
there is no action or store change after it so no need for a snapshot. the snapshot that was here before was the same as the last event's snapshot

**Review by amirfefer — DISMISSED** [review]
Tested for report template page and it works well !
one important thing left (separate PR) is to add an`id` for enabling multiple autocomplete inputs in a page

**Comment by Ron-Lavi** [comment]
Thanks @amirfefer,
added a PR for that here: https://github.com/theforeman/foreman/pull/6501/commits/b1384c41f0c5c4837ce1255fe75d23b408abfd45

**Comment by Ron-Lavi** [comment]
@sharvit @tbrisker could you guys have a look? this is part of the code that you reviewed in `replacing form-autocomplete`, thanks :)

**Comment by amirfefer** [comment]
I found a little regression when I consumed this component, the clearing `x`'s position looks odd:
report tempalte

**Comment by Ron-Lavi** [comment]
@amirfefer just added a fix if you could have a look, thanks :)

**Comment by Ron-Lavi** [comment]
test failures seem unrelated

**Review by amirfefer — CHANGES_REQUESTED** [review]
Thanks @Laviro , looks better, but still seems that there are some css issue with the `X` clear button's position:
auto-complete-1

**Comment by Ron-Lavi** [comment]
Thanks @amirfefer, on my stack it worked fine..
but I guess some other styling are affecting this, i will check into your PR to see where the problem is,
but maybe meanwhile we can merge this one - and fix the styling issue on a follow up PR?
as it doesn't affect the SearchBar which is the only consumer of this component at the moment.

**Comment by amirfefer** [comment]
@Laviro  how about creating a storybook for a standalone `autocomplete`

**Comment by Ron-Lavi** [comment]
This is how it looks on my side when I switch the SearchBar with the Autocomplete:
architectures

**Comment by Ron-Lavi** [comment]
@amirfefer, I added a storybook - and I do see some styling problems,
Not sure if i will get to it any time soon. though I think we can really make it to a follow up PR,
as the main idea of braking the `form-autocomplete` PR was to have small PRs.

**Review by amirfefer — APPROVED** [review]
Thanks @Laviro 
I'm in favor of merging this now, while css fixing could be in #6531

**Comment by amirfefer** [comment]
@ohadlevy can we merge this  (and #6501)

**Review by ares — APPROVED** [review]
Thanks @Laviro, I didn't see any regression while testing. Merging! Test failures are unrelated.
